### DISCUSSION TITLE
Next js build and deploy error

### DISCUSSION THREAD

**Discussion by aly8i** [vercel/community#1003] [Help] [answered]
The next js application crashes when I run `npm run build` then `npm start`. However, it works fine using `npm run dev`.

here is a youtube video for me illustrating my problem: [click here](https://www.youtube.com/watch?v=hsVwZvslfTM)

here is the github repository for my project : [click here](https://github.com/aly8i/resturant)

to install dependencies: `npm install --force`
to run the server: `cd backend` `cd auth` `npm start`
to run the application: `npm run dev`

my problem commands: `npm run build` `npm run start`

note: all my .env are in the repository so the application will work immediately without configuration, please don't miss use them.

**Comment by aly8i** [comment]
@amyegan please check this out.

**Comment by amyegan** [comment, marked as answer]
Hello again @aly8i! I'm not seeing `req.user` come through in the login function and there is nothing to handle the scenario where this data is missing. That results in the application waiting indefinitely for a response that will never come, and it eventually errors out. 

Have you tried adding logging to your Serverless Function and backend service? If you add some logging and error handling on these routes, I think it will be easier to see what data is missing and find the point where it is breaking.
